### PR TITLE
Implement event tests

### PR DESCRIPTION
**Pull request by mijatkrivokapic**
Implemented unit, integration and e2e tests for event management.

- Key Changes
  - EventServiceTest.java: 
    - Comprehensive Test Cases: Added new test cases for event creation, update, and deletion scenarios, including edge cases like past dates, non-existent event types/organizers, and events with reservations.
    - Publicity Logic Validation: Enhanced tests to verify correct behavior when changing event publicity (private to public, public to private), ensuring guest lists are cleared and stats are reset appropriately.
    - Notification Verification: Included tests to confirm that notifications are sent to accepted guests upon event updates or cancellations.
    - Agenda Item Management: Introduced detailed test cases for creating, updating, and deleting agenda items, covering validation of start/end times, existence checks, and proper association with events.
    - Agenda Retrieval & Sorting: Added tests for getAgenda to ensure correct filtering of deleted items and sorting by start time.
    - Helper Comparison Methods: Refined compareEvents, compareEventStats, and compareAgendaItems for more accurate object comparison in assertions.
  - EventTest.java (Selenium UI Tests):
    - Improved Form Validation Tests: Expanded tests for create and edit event forms to cover scenarios where fields are empty or dates are in the past, verifying that the submit button is appropriately disabled.
    - Event Type Handling: Added tests to confirm event creation and update work correctly when "No event types apply" is selected.
    - Agenda Item UI Interaction: Introduced new UI tests for creating, editing, and deleting agenda items directly through the web interface, including form validation and confirmation dialogs.
    - Enhanced Assertions: Utilized helper methods (verifyEventDetails, verifyAgendaItem) for more concise and readable assertions of UI elements.
  - EventControllerTest.java:
  - Authorization Tests: Strengthened authorization tests across all endpoints (create, update, delete events, and agenda items) for EVENT_ORGANIZER, ADMIN, and AUTHENTICATED_USER roles.
  - Error Handling Scenarios: Added more specific test cases for various error conditions, such as non-existent IDs, invalid input formats, and business rule violations (e.g., past dates, start time after end time).

### CODE REVIEW

File: eventPlanner/src/test/java/com/ftn/iss/eventPlanner/service/EventServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,1343 @@
...
+        CreatedEventDTO result = eventService.create(createEventDTO);
+
+        verify(eventRepository,times(1)).save(eventCaptor.capture());
+        Event savedEvent = eventCaptor.getValue();
+
+        assertTrue(compareEvents(savedEvent, createdEvent));
+    }
+
+    @Test
+    @DisplayName("Should throw exception when EventType is not found")
```
Comment (milicaradicc) [inline eventPlanner/src/test/java/com/ftn/iss/eventPlanner/service/EventServiceTest.java:407]: You can add test for when event type is not active

File: eventPlanner/src/test/java/com/ftn/iss/eventPlanner/service/EventServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,1343 @@
...
+
+        verify(eventStatsRepository, times(1)).save(eventStatsCaptor.capture());
+        EventStats savedStats = eventStatsCaptor.getValue();
+
+        assertTrue(compareEventStats(savedStats, updatedStats));
+        assertTrue(compareEvents(savedEvent, updatedEvent));
+    }
+
+    @Test
+    @DisplayName("Should throw exception when event doesn't exist")
```
Comment (milicaradicc) [inline eventPlanner/src/test/java/com/ftn/iss/eventPlanner/service/EventServiceTest.java:502]: You can add test for when event is deleted

**Review by milicaradicc — CHANGES_REQUESTED** [review]
Great job – the tests are very detailed and passing successfully. Please make sure to also add the tests mentioned in the comments

**Review by milicaradicc — DISMISSED** [review]

**Review by milicaradicc — APPROVED** [review]
